static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,573 @@
+    if (!photometry) {
+      dispatch(photometryActions.fetchSourcePhotometry(id));
+    }
+    if (photometry) {
+      setInstruments([...new Set(photometry.map((x) => x.instrument_name))]);
+      setFilters([...new Set(photometry.map((x) => x.filter))]);
+    }
+
+    if (photometry && !run) {
+      const data = photometry.filter(
+        (x) =>
+          x.filter === params.filter && x.instrument_name === params.instrument
+      );
+      const times = data.map((x) => x.mjd);
+      const mag = data.map((x) => x.mag);
+      const magerr = data.map((x) => x.magerr);
+      if (!plotted) {
+        plotdata(times, mag, magerr);
+        setPlotted(true);
+      }
+
+      // filtered times
+      const ftimes = times.filter((x) => x >= mjdmin && x <= mjdmax);
+      const fmag = mag.filter(
+        (x, i) => times[i] >= mjdmin && times[i] <= mjdmax
+      );
+      const fmagerr = magerr.filter(
+        (x, i) => times[i] >= mjdmin && times[i] <= mjdmax
+      );
+
+      const gls = GLS(ftimes, fmag, {
+        e_y: fmagerr,
+        ofac: parseInt(params.ofac, 10),
+        fbeg: null,
+        fend: null,
+        ls: null,
+      });
+      plotGLS(gls.f, gls.p, {
+        periodbest: 1 / gls.fbest,
+        periodmax: gls.tbase / 3.5,
+      });
+      setBestp(1 / gls.fbest);
+      setRun(true);
+    }
+    if (run) {
+      const data = photometry.filter(
+        (x) =>
+          x.filter === params.filter &&
+          x.instrument_name === params.instrument &&
+          x.mjd >= mjdmin &&
+          x.mjd <= mjdmax
+      );
+      const times = data.map((x) => x.mjd);
+      const mag = data.map((x) => x.mag);
+      const title = `P=${bestp?.toFixed(8)} d (${params.instrument}: ${
+        params.filter
+      })`;
+      plotphased(times, mag, bestp, title);
+    }
+  }, [id, photometry, bestp, run, dispatch]);
+
+  const componentRef = useRef();
+
+  const initialFormState = {
+    ...params,
+  };
+
+  const onSubmit = () => {
+    const formData = {
+      ...initialFormState,
+      ...getValues(),
+    };
+    setPlotted(false);
+    setRun(false);
+    setParams(formData);
+  };
+
+  const rules = { required: true, min: 1, max: 25, type: "number", step: 1 };
+
+  return (
+    <>
+      <div>
+        <div>
+          <Typography variant="h5" gutterBottom>
+            Interactive Periodogram for{" "}
+            <Link to={`/source/${id}`} role="link">
+              {id}
+            </Link>
+          </Typography>
+        </div>
+        <Grid
+          container
+          direction="row"
+          justify="flex-start"
+          alignItems="flex-start"
+          spacing={1}
+        >
+          <Grid item xs={10}>
+            <Card>
+              <CardContent ref={componentRef}>
+                <div>
+                  {photometry ? (
+                    <div ref={dataplotRef} className={classes.dygraphChart} />
+                  ) : (
+                    <div>{placeholder}</div>
```

Comment:
See comment above - prefer to just move the JSX block inline here

Reply:
Done.